manifests/init.pp

```diff
@@ -324,6 +324,15 @@
 
   $template_snmpd_conf          = 'snmp/snmpd.conf.erb'
   $template_snmpd_sysconfig     = "snmp/snmpd.sysconfig-${facts['os']['family']}.erb"
+  case $::operatingsystemrelease {
```

Comment:
please don't use the legacy facts. The modern way is to use the facts hash. Probably `$facts['os']['release']['full']` if what you want.

Reply:
Clena you please use the [selector pattern](https://puppet.com/docs/puppet/5.3/lang_conditional.html#selectors) instead?